core/random/concurrentSafeIntRandomizer_test.go

```diff
@@ -22,28 +22,38 @@ func TestConcurrentSafeIntRandomizer_IntnConcurrent(t *testing.T) {
 	for i := 0; i < maxIterations; i++ {
 		go func(idx int) {
 			for {
-				_, err := csir.Intn(idx + 1)
-				assert.Nil(t, err)
-
+				_ = csir.Intn(idx + 1)
 				time.Sleep(time.Millisecond)
 			}
 		}(i)
 	}
 
-	fmt.Println("Waiting 10 seconds...")
-	time.Sleep(time.Second * 10)
+	fmt.Println("Waiting 1 second...")
+	time.Sleep(time.Second * 1)
 }
 
-func TestConcurrentSafeIntRandomizer_IntnInvalidParamShouldPanic(t *testing.T) {
-	defer func() {
-		r := recover()
+func TestConcurrentSafeIntRandomizer_IntnInvalidShouldReturnZero(t *testing.T) {
+	t.Parallel()
 
-		assert.NotNil(t, r)
-	}()
 	csir := &ConcurrentSafeIntRandomizer{}
 	assert.False(t, csir.IsInterfaceNil())
 
-	res, err := csir.Intn(-1)
+	res := csir.Intn(-1)
 	assert.Equal(t, 0, res)
-	assert.NotNil(t, err)
+
+	res = csir.Intn(0)
```

Comment:
You can add another assert also for 1:
	res = csir.Intn(1)
	assert.Equal(t, 0, res)

Reply:
done